addon/webextension/background/selectorLoader.js

```diff
@@ -1,78 +1,134 @@
-/* globals browser, catcher */
-window.selectorLoader = (function () {
+/* globals browser, catcher, log */
+
+"use strict";
+
+var global = this;
+
+this.selectorLoader = (function() {
   const exports = {};
-  // These modules are loaded in order, and some need to be listed before others
-  // due to dependencies:
-  const scripts = [
+
+  // These modules are loaded in order, first standardScripts, then optionally onboardingScripts, and then selectorScripts
+  // The order is important due to dependencies
+  const standardScripts = [
+    "build/buildSettings.js",
+    "log.js",
     "catcher.js",
+    "assertIsTrusted.js",
     "background/selectorLoader.js",
     "selector/callBackground.js",
+    "selector/util.js"
+  ];
+
+  const selectorScripts = [
     "clipboard.js",
     "makeUuid.js",
     "build/shot.js",
     "randomString.js",
     "domainFromUrl.js",
     "build/inlineSelectionCss.js",
     "selector/documentMetadata.js",
-    "selector/util.js",
     "selector/ui.js",
     "selector/shooter.js",
     "selector/uicontrol.js"
   ];
 
-  exports.unloadIfLoaded = function () {
-    return browser.tabs.executeScript({
-      code: "window.selectorLoader && window.selectorLoader.unloadModules()"
+  // These are loaded on request (by the selector worker) to activate the onboarding:
+  const onboardingScripts = [
+    "build/onboardingCss.js",
+    "build/onboardingHtml.js",
+    "onboarding/slides.js"
+  ];
+
+  exports.unloadIfLoaded = function(tabId) {
+    return browser.tabs.executeScript(tabId, {
+      code: "this.selectorLoader && this.selectorLoader.unloadModules()",
+      runAt: "document_start"
+    }).then(result => {
+      return result && result[0];
+    });
+  };
+
+  exports.testIfLoaded = function(tabId) {
+    if (loadingTabs.has(tabId)) {
+      return true;
+    }
+    return browser.tabs.executeScript(tabId, {
+      code: "!!this.selectorLoader",
+      runAt: "document_start"
     }).then(result => {
-      return result && result.toString() === "true";
+      return result && result[0];
+    });
+  };
+
+  let loadingTabs = new Set();
+
+  exports.loadModules = function(tabId, hasSeenOnboarding) {
+    let promise;
+    loadingTabs.add(tabId);
+    if (hasSeenOnboarding) {
+      promise = executeModules(tabId, standardScripts.concat(selectorScripts));
+    } else {
+      promise = executeModules(tabId, standardScripts.concat(onboardingScripts).concat(selectorScripts));
+    }
+    return promise.then((result) => {
+      loadingTabs.delete(tabId);
+      return result;
+    }, (error) => {
+      loadingTabs.delete(tabId);
+      throw error;
     });
   };
 
-  exports.loadModules = function () {
+  function executeModules(tabId, scripts) {
     let lastPromise = Promise.resolve(null);
     scripts.forEach((file) => {
       lastPromise = lastPromise.then(() => {
-        return browser.tabs.executeScript({file})
+        return browser.tabs.executeScript(tabId, {
```

Comment:
(clarified to refer to async/await) – we haven't used async/await at all yet in the codebase, that's a refactoring I'd like to put off until a later date with less time pressure